Sources/Purchasing/StoreKitAbstractions/StoreProductDiscount.swift

```diff
@@ -70,7 +77,13 @@ public final class StoreProductDiscount: NSObject, StoreProductDiscountType {
     @objc public var offerIdentifier: String? { self.discount.offerIdentifier }
     @objc public var currencyCode: String? { self.discount.currencyCode }
     // See also `priceDecimalNumber` for Objective-C
-    public var price: Decimal { self.discount.price }
+    public var price: Decimal {
+        // See: https://github.com/RevenueCat/purchases-ios/issues/2207
+        // it seems like StoreKit 2 doesn't actually round decimals in a price-friendly fashion, so we do it
+        let priceNSDecimal = (self.discount.price as NSDecimalNumber)
```

Comment:
ended up going back and forth to objc here, I bet there's a nicer solution

Reply:
Hmmm yeah let me look at these.